docs/TransparentLogging.md

```diff
@@ -0,0 +1,280 @@
+   to be machine-checkable (but need not be strict – see below).
+ - Authentication: The submissions are signed by one of a limited number of
+   private keys, so the signature can be checked with the corresponding public
+   key.  The Log is thus implicitly
+     - assuming that the holders of those private keys will not sign excessive
+       numbers of objects (and so allow the Log to be swamped)
+     - trusting the key holders not to sign anything that might be problematic
+       for the Log to host.
+
+For our examples:
+ - A Certificate Transparency log requires entries to have X.509 structure, and
+   trusted-roots authentication.
+ - A Binary Transparency log requires entries to have the structure of an
+   installable binary package, and vendor-signing key authentication.
+ - A Pastebin log has little structural requirement and no authentication
+   requirement, and so is vulnerable to different attacks (swamping, uploading
+   illegal content etc.)
+
+Finally, note that a transparent Log normally acts as an **observatory, not as a
+police officer**, in the overall ecosystem.  With this in mind, it's often
+sensible for the structure checks on submissions to be lax, so that
+technically-invalid objects that are still signed and distributed can be
+monitored and attributed.
+
+ - A Certificate Transparency Log may well accept X.509 certificates that fail
+   various sorts of validity check, as long as the signature validation is
+   correct.  This means that certificates that might be accepted by lax clients
+   are still logged, and certificate authorities that issue invalid certificates
+   can be held to account more easily.
+
+## Inclusion Proofs vs. Promises
+
+Transparent Logs have two distinct mechanisms for guaranteeing inclusion of a
+particular entry in the Log.
+
+ - An **inclusion proof** gives cryptographic proof that a particular Merkle
+   tree head includes a given entry: the entry's hash and the proof hashes can be
+   combined to recreate the root hash of the tree.
+ - An **inclusion promise** is a signed commitment from the Log to incorporate a
+   particular entry within a defined time period.  This requires a separate
+   subsequent check to confirm that the Log does indeed fulfil this promise.
+
+A related factor is the number of entries that are incorporated into each new
+tree head issued by the Log, which we'll call the tree head **batch size**.
+
+ - Logs that issue inclusion promises can batch together multiple submissions
+   (e.g. an hour's worth) into a new tree head.  This is more likely to produce
+   large batch sizes, but is not guaranteed to (e.g. if submission rates are
+   very low).
+ - Logs that immediately issue inclusion proofs are likely to have smaller batch
+   sizes, either because they issue a new tree head per submission (batch size =
+   1), or because the amount of batching possible while still responding to
+   submissions quickly is small.
```

Comment:
There's a 2.5 option too, where the log responds either with "retry in x", or the inclusion proof, and then internally queues and batches as it sees fit. 
Technically I guess this is like (1) with polling of get-sth and get-proof-by-hash, but might be worth mentioning to seed the thought in anyone designing a new system which would prefer larger batches and could tolerate a delay in integration.

Reply:
I've reworded a smidgeon to not rule this out, but I don't think this doc is the place to go any deeper on this...